### PR TITLE
pythonPackages.textacy: init at 0.4.1

### PR DESCRIPTION
**Pull request by rvl**
#### Motivation for this change

Natural language processing with Python under Nix.

* This change adds the textacy python package and its dependencies.
* Versions have been tweaked on some dependencies where necessary.
* Finally, this change also adds python package derivations for the language model datasets required for many NLP functions.

###### Datasets

The language data models can't be downloaded in the way described by the spaCy docs because their download script is a wrapper around `pip install`. The data files total a couple GB and need enough `/tmp` space to assemble. I'm assuming that because the derivations exist under an attrset, they won't be automatically built by Hydra (good).

#### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"` -- didn't use nox but have tested build of `mopidy-gmusic`, but not the "oslo" packages because they are already marked broken.
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

/cc @sdll

### CODE REVIEW

File: pkgs/development/python-modules/pyemd/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ stdenv
+, pkgs
```
Comment (FRidh) [inline pkgs/development/python-modules/pyemd/default.nix]: no pkgs

File: pkgs/development/python-modules/spacy/models.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ stdenv
+, pkgs
```
Comment (FRidh) [inline pkgs/development/python-modules/spacy/models.nix]: no pkgs

File: pkgs/development/python-modules/textacy/default.nix [hunk 3]
```diff
@@ -0,0 +1,128 @@
+{ stdenv
+, pkgs
```
Comment (FRidh) [inline pkgs/development/python-modules/textacy/default.nix]: no pkgs

File: pkgs/development/python-modules/textacy/default.nix [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+, matplotlib
+, cython
+, ftfy
+, pyemd
+, six
+, cffi
+, nose
+}:
+let
+  ijson = buildPythonPackage rec {
```
Comment (FRidh) [inline pkgs/development/python-modules/textacy/default.nix]: why are these packages not called from `python-packages.nix`?

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -14313,6 +14313,20 @@ in {
      };
    };
 
+   cachetools = buildPythonPackage rec {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: separate file in `python-modules/`

File: pkgs/top-level/python-packages.nix [hunk 6]
```diff
@@ -24150,11 +24164,11 @@ EOF
 
 
   unidecode = buildPythonPackage rec {
-    name = "Unidecode-0.04.18";
+    name = "Unidecode-0.04.21";
 
     src = pkgs.fetchurl {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: fetchPypi and preferable separate file

File: pkgs/top-level/python-packages.nix [hunk 7]
```diff
@@ -27308,6 +27322,12 @@ EOF
 
   spacy = callPackage ../development/python-modules/spacy { };
 
+  spacy_models = callPackage ../development/python-modules/spacy/models.nix { };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: Might be nicer to add this to `passthru.models` of `spacy`. No need though.

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by rvl** [comment]
Thanks for the review @FRidh. I have addressed your comments in new commits which can be autosquashed.

File: pkgs/top-level/python-packages.nix (hunk 7 above)
Comment (rvl) [reply pkgs/top-level/python-packages.nix]: I think I will pass on this because the data model packages have dependencies on spacy.

**Comment by FRidh** [comment]
@rvl please do the squashing yourself; commit per package.

**Comment by rvl** [comment]
@FRidh OK, have pushed the squashed commits.

**Comment by rvl** [comment]
Thanks @FRidh :-)

**Comment by rvl** [comment]
Very sorry @FRidh, it appears I forgot to retest the spacy_models packages. Would you be able to apply this patch? https://github.com/rvl/nixpkgs/commit/b7e602a9736db07a64b8bea1dc4d9319606206de

**Pull request merged**